serverlessworkflow/modules/ROOT/pages/service-orchestration/orchestration-of-openapi-based-services.adoc

```diff
@@ -0,0 +1,423 @@
++
+.Running the Subtraction application
+[source,shell,subs="attributes+"]
+----
+$ java -Dquarkus.http.port=8181 -jar subtraction-service/target/quarkus-app/quarkus-run.jar
+----
+
+4. In another terminal, run the Multiplication application from the project's parent directory.
++
+.Running the Multiplication application
+[source,shell,subs="attributes+"]
+----
+$ java -Dquarkus.http.port=8282 -jar multiplication-service/target/quarkus-app/quarkus-run.jar
+----
+
+5. Now that you have both services running, run the workflow application in a new terminal.
++
+.Running the Serverless Workflow application
+[source,shell,subs="attributes+"]
+----
+$ java -jar multiplication-service/target/quarkus-app/quarkus-run.jar
+----
+
+6. You can submit a new request to the running workflow in a new terminal.
++
+.Making a request to the Serverless Workflow application
+[source,shell,subs="attributes+"]
+----
+$ curl -X POST \
+    -H 'Content-Type:application/json' \
+    -H 'Accept:application/json' \
+    -d '{"workflowdata" : { "fahrenheit": 100  } }' \
+    http://localhost:8080/fahrenheit_to_celsius
+----
++
+You can see the result of all the calls made by the workflow during this execution:
++
+.Reponse from the workflow execution
+[source,json]
+----
+{
+  "id": "35d8bede-a621-4e18-83e6-6d7ccfb4f650",
+  "workflowdata": {
+    "fahrenheit": 100, <1> 
+    "subtractValue": 32, <2>
+    "multiplyValue": 0.5556, <3>
+    "difference": 68, <4> 
+    "product": 37.7808 <5>
+  }
+}
+----
++
+<1> The data input
+<2> The constant value of the conversion formula
+<3> The constant value of the conversion formula
+<4> The result of the call to the Subtraction Service
+<5> The result of the call to the Multiplication Service
++
+Alternatively to `curl`, you can use the link:http://localhost:8080/q/swagger-ui/#/Fahrenheit%20_to%20_celsius%20Resource/createResource_fahrenheit_to_celsius[OpenAPI Swagger UI] to make calls to the workflow application.
+
+== Configuration of the OpenAPI Services
+
+Internally, Kogito uses the link:{quarkus_openapi_gen_url}[Quarkus OpenAPI Generator extension] and the link:{quarkus_rest_client_url}[Quarkus REST Client]. All the configurations are based on these two components backed up by link:{quarkus_config_url}[Quarkus Configuration].
+
+To configure the endpoints, you must use the sanitized OpenAPI specification file name as the link:{quarkus_rest_client_url}#create-the-configuration[REST client configuration key]. For example, for a file named `subtraction.yaml`, the configuration key is `subtraction_yaml`.
+
+[TIP]
+====
+For more information about how Kogito sanitizes the file name, see link:{mp_config_env_vars_url}[Environment Variables Mapping Rules]. The key must be a valid environment variable.
+====
+
+This configuration key is used on every property related to the Quarkus REST Client. For example, to configure the URL, you can use:
+
+.Example of URL configuration
```

Comment:
This now seems like optional configuration, but specifying the url property is a requirement, right?  Shouldn't this be like the 3rd step in the procedure more (first get openapi definition, next invoke function in workflow, third configure url (and optionally maybe other config options))?

Also, since we know this url property will raise some questions (around hardcoding a url) and we want to evolve this with better service discovery in the future, maybe it's worth having a small paragraph to highlight this future work?

Reply:
The linked guide (Configuring OpenAPI services endpoints) will do just that. It will explain how to properly configure the endpoints, not to keep them hardcoded in the config. But I think I can write a word about environment variables replacement in the app properties and the future svc discovery feature in the roadmap.